aiida_quantumespresso/parsers/pp.py

```diff
@@ -0,0 +1,332 @@
+                    'DEPRECATED:': None,
+                }
+            }
+
+            # Match any known error and warning messages
+            for marker, message in message_map['error'].items():
+                if marker in line:
+                    if message is None:
+                        message = line
+                    logs.error.append(message)
+
+            for marker, message in message_map['warning'].items():
+                if marker in line:
+                    if message is None:
+                        message = line
+                    logs.warning.append(message)
+
+
+        stdout_lines = stdout_str.splitlines()
+        logs = get_logging_container()
+        output_dict = {}
+
+        # Check for job completion, indicating that pp.x exited without interruption, even if there was an error.
+        for line in stdout_lines:
+            if 'JOB DONE' in line:
+                break
+        else:
+            logs.error.append('ERROR_OUTPUT_STDOUT_INCOMPLETE')
+
+        # Detect any issues and detect job completion
+        for line in stdout_lines:
+            detect_important_message(logs, line)
+
+        # Parse useful data from stdout
+        for line in stdout_lines:
+            if 'Check:' in line:
+		if 'negative/imaginary' in line:    # QE6.1
+                    output_dict['negative_core_charge'] = float(line.split('=')[-1].split()[0])
+                    output_dict['imaginary_core_charge'] = float(line.split('=')[-1].split()[-1])
+		else:                               # QE6.4
+                    output_dict['negative_core_charge'] = float(line.split('=')[1])
+            if 'Min, Max, imaginary charge:' in line:
+                output_dict['charge_min'] = float(line.split()[-3])
+                output_dict['charge_max'] = float(line.split()[-2])
+                output_dict['charge_img'] = float(line.split()[-1])
+            if 'plot_num = ' in line:
+                output_dict['plot_num'] = int(line.split('=')[1])
+            if 'Plot Type:' in line:
+                output_dict['plot_type'] = line.split('Output format')[0].split(':')[-1].strip()
+                output_dict['output_format'] = line.split(':')[-1].strip()
+
+        return logs, output_dict
+
+
+    def parse_gnuplot1D(self, data_file_str):
+        """
+        Parse 1D GNUPlot formatted output
+
+        :param data_file_str: the data file read in as a single string
+        """
+        data_lines = data_file_str.splitlines()
+
+        n_col = len(data_lines[0].split())
+
+        # 1D case
+        if n_col == 2:
+            coords = []
+            data = []
+            data_integral = []
+            for line in data_lines:
+                split_line = line.split()
+                coords.append(float(split_line[0]))
+                data.append(float(split_line[1]))
+            y_data_lists = [data]
+            y_array_names = ['data']
+            y_array_units = ['']
```

Comment:
is there no way of knowing the units?

Reply:
AFAIK the units are determined by the input `plot_num` (different units for different kinds of calculations) - so we would just need a lookup table.

Reply:
Units are now looked up from a dictionary.